### PR TITLE
execution: misc cleanups around parameter passing

### PR DESCRIPTION
**Pull request by MichaHoffmann**

### CODE REVIEW

File: execution/aggregate/hashaggregate.go [hunk 1]
```diff
@@ -125,20 +120,15 @@ func (a *aggregate) Next(ctx context.Context) ([]model.StepVector, error) {
...
-		}
-		for i := range a.params {
-			a.params[i] = math.NaN()
-			if i < len(args) && len(args[i].Samples) > 0 {
-				a.params[i] = args[i].Samples[0]
-				a.paramOp.GetPool().PutStepVector(args[i])
-			}
-		}
-		a.paramOp.GetPool().PutVectors(args)
+	args, err := a.paramOp.Next(ctx)
```
Comment (fpetkovski) [inline execution/aggregate/hashaggregate.go:123]: Would this cause extra allocations for aggregations without a parameter?

File: engine/explain_test.go [hunk 2]
```diff
@@ -66,6 +66,9 @@ func TestQueryExplain(t *testing.T) {
 					{
 						OperatorName: "[concurrent(buff=2)]", Children: []engine.ExplainOutputNode{
 							{OperatorName: "[aggregate] sum by ([job])", Children: []engine.ExplainOutputNode{
+								{
+									OperatorName: "[noop]",
+								},
```
[the remark this answers is not in the document]
Comment (MichaHoffmann) [reply engine/explain_test.go:72]: I mean its the engine reporting accurately what the engine is doing! But we can probably hide it in UI

File: execution/aggregate/hashaggregate.go (hunk 1 above)
Comment (MichaHoffmann) [reply execution/aggregate/hashaggregate.go:123]: Didnt see a difference in benchmarks!

File: engine/explain_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fpetkovski) [reply engine/explain_test.go:72]: I think it's better to hide it from this operator's Explain function, maybe through a switch on the aggregation type. We can also have a separate constructor for aggregations with a parameter and the noop can be created internally.

[the remark this answers is not in the document]
Comment (MichaHoffmann) [reply engine/explain_test.go]: Ok; i only add the param op if we use a aggregation or function that has a param now!

File: execution/aggregate/hashaggregate.go (hunk 1 above)
Comment (MichaHoffmann) [reply execution/aggregate/hashaggregate.go:128]: Ill resolve this really quick

**Review by fpetkovski — APPROVED** [review]

**Pull request merged**